assets/app/scripts/controllers/builds.js

```diff
@@ -21,6 +22,9 @@ angular.module('openshiftConsole')
     var watches = [];
 
     watches.push(DataService.watch("builds", $scope, function(builds, action, build) {
+      var buildConfigName;
+      var buildName;
```

Comment:
We can, but I'd love to chat about it a bit more.  The variable hoisting in JS isn't something I'm crazy about trusting.  I'd probably prefer we initialize it to an empty string or matching type, then set it to a different value in the if statement:

```javascript
watches.push(DataService.watch("builds", $scope, function(builds, action, build) {
      var buildConfigName = '';
      var buildName = '';
      // ..... other code
      if (build) {
        buildConfigName = build.metadata.labels.buildconfig;
        buildName = build.metadata.name;
      }
```
This is a little better because the vars at least have a value right away.  Stylistically I do like the habit of declaring all vars at the top of every function in that it makes it consistently clear what functions will manipulate, but the real problem here is letting JavaScript move the variables behind the scenes...